crates/core/component/dex/src/state_key.rs

```diff
@@ -237,5 +246,11 @@ pub(crate) mod eviction_queue {
 
             full_key
         }
+
+        pub(crate) fn parse_id_from_key(key: Vec<u8>) -> anyhow::Result<[u8; 32]> {
+            ensure!(key.len() == 155, "key must be 155 bytes");
+            let k = &key[123..155];
+            Ok(k.try_into()?)
+        }
```

Comment:
Slightly gnarly, I'll admit. This save us 32 bytes of nonverifiable storage, which actually kind of adds up to `64Kb` per trading pair per block in the pathological case.